modules/viewing-the-image-pull-source.adoc

```diff
@@ -0,0 +1,44 @@
+// Module included in the following assemblies:
+//
+// *installing/validating-an-installation.adoc
+
+[id="viewing-the-image-pull-source_{context}"]
+= Viewing the image pull source
+
+You can view the source of your pulled images by using the CRI-O logs on your nodes. 
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+* Review the CRI-O logs on a node or for the worker nodes:
++
+[source,terminal]
+----
+$  oc adm node-logs ci-ln-jb5m1vb-f76d1-cfzck-master-0 -u crio
+----
++
+.Example output
++
+The `Trying to access` log entry indicates where the images are being pulled from.
++
+[source,terminal]
+----
+...
+Aug 05 10:33:21.594996 ip-10-0-138-140.ec2.internal crio[1366]: time="2021-08-05 10:33:21.594930907Z" level=info msg="Pulling image: quay.io/openshift-release-dev/ocp-release:4.8.4-ppc64le" id=abcd713b-d0e1-4844-ac1c-474c5b60c07c name=/runtime.v1alpha2.ImageService/PullImage
+Aug 05 10:33:21.597931 ip-10-0-138-140.ec2.internal crio[1366]: time="2021-08-05 10:33:21.597839071Z" level=info msg="Trying to access \"quay.io/openshift-release-dev/ocp-release:4.8.4-ppc64le\""
+...
+----
+
++
+[NOTE]
+====
+For disconnected installations, if configured correctly, even though the image is pulled from the mirrored location: 
+
+* To view the source of pulled images, you must review the `Trying to access` log entry in the CRI-O logs. Other methods to view the image pull source, such as using the `crictl images` command on a node, show the full name of the image.  For example: `quay.io/project/image-name`. 
+
+* If your cluster uses an `ImageContentSourcePolicy` object to configure repository mirroring, the CRI-O logs report the full name of the image repository. For example: `quay.io/project/image-name`. 
```

Comment:
I think this description is not correctly, If the cluster create an `ImageContentSourcePolicy` object to configure repository mirroring, the crio log should report mirror registry name of the image. 
@QiWang19  Can you double confirm?

Reply:
@lyman9966 @QiWang19 This statement came from @umohnani8 [comment in a related BZ](https://bugzilla.redhat.com/show_bug.cgi?id=1881694#c7) and is shown in [your example in that BZ](https://bugzilla.redhat.com/show_bug.cgi?id=1881694#c17).
Seeing `"Pulled image: quay.io/openshift-release-dev/...` would confuse me as a user. I wanted to clarify that this line is present in the output, but does not indicate where the image was actually pulled from. Please correct me if I am misunderstanding.

Reply:
So if using an ICSP where you are mirroring `quay.io/myrepo` to `localreg/myrepo`, when you pull down `quay.io/myrepo/myiage` and do crictl images, the image will show as `quay.io/myrepo/myimage`, the original name will be used everywhere.
However, if you want to see where the image was actually pulled from, you can look at the cri-o logs and it should show something like `Pulled image: localreg/myrepo/myimage`.
Hopefully this makes sense, it can be confusing for users as they think their mirrors are not working when they actually are.

Reply:
@umohnani8 If I understand, the output in the BZ that shows `"Pulled image: quay.io/openshift-release-dev/...` is from the CRI-O logs that verifies that the image is pulled from the mirror repo: 

```
crio logs as below:
Mar 17 02:52:50 minmli0317gcp1-46m6l-m-0.c.openshift-qe.internal crio[1484]: time="2021-03-17 02:52:50.178706830Z" level=info msg="Checking image status: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:1926eae7cacb9c00f142ec98b00628970e974284b6ddaf9a6a086cb9af7a6c31" id=d71a23f1-17e0-40d0-b967-57f2e474bf0c name=/runtime.v1alpha2.ImageService/ImageStatus
Mar 17 02:52:50 minmli0317gcp1-46m6l-m-0.c.openshift-qe.internal crio[1484]: time="2021-03-17 02:52:50.179161853Z" level=info msg="Image quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:1926eae7cacb9c00f142ec98b00628970e974284b6ddaf9a6a086cb9af7a6c31 not found" id=d71a23f1-17e0-40d0-b967-57f2e474bf0c name=/runtime.v1alpha2.ImageService/ImageStatus
Mar 17 02:52:50 minmli0317gcp1-46m6l-m-0.c.openshift-qe.internal crio[1484]: time="2021-03-17 02:52:50.190753958Z" level=info msg="Pulling image: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:1926eae7cacb9c00f142ec98b00628970e974284b6ddaf9a6a086cb9af7a6c31" id=f740d165-4482-439f-aced-77c781f33731 name=/runtime.v1alpha2.ImageService/PullImage
Mar 17 02:52:50 minmli0317gcp1-46m6l-m-0.c.openshift-qe.internal crio[1484]: time="2021-03-17 02:52:50.194341109Z" level=info msg="Trying to access \"minmli0317gcp1.mirror-registry.qe.gcp.devcluster.openshift.com:5000/ocp/release@sha256:1926eae7cacb9c00f142ec98b00628970e974284b6ddaf9a6a086cb9af7a6c31\""
Mar 17 02:52:50 minmli0317gcp1-46m6l-m-0.c.openshift-qe.internal crio[1484]: time="2021-03-17 02:52:50.226788351Z" level=info msg="Trying to access \"minmli0317gcp1.mirror-registry.qe.gcp.devcluster.openshift.com:5000/ocp/release@sha256:1926eae7cacb9c00f142ec98b00628970e974284b6ddaf9a6a086cb9af7a6c31\""
Mar 17 02:52:52 minmli0317gcp1-46m6l-m-0.c.openshift-qe.internal crio[1484]: time="2021-03-17 02:52:52.936392518Z" level=info msg="Pulled image: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:1926eae7cacb9c00f142ec98b00628970e974284b6ddaf9a6a086cb9af7a6c31" id=f740d165-4482-439f-aced-77c781f33731 name=/runtime.v1alpha2.ImageService/PullImage
```

Reply:
@mtrmac does this log look correct? When pulling an image from a mirror it just logs `Trying to access:...` but not the fact that it actually pulled from thee? Would there be a `failed to access` if it couldn't access the mirror location?

Reply:
If it successfully accessed a mirror, can we log that it actually pulled the image from there in c/image? We moved these logs from debug level to info level so users can see where the image is pulled from as the image having the same name as the original location was causing confusion where users thought that their images weren't being pulled from the mirror.
@QiWang19 I remember you changed the log level, any idea if we can add a log of where it actually pulled the image from after the `Trying to access:...` logs?

Reply:
Eventually there’s _some_ concern about too many logs, but this looks resonable, especially with the existing `DockerLogMirrorChoice` opt-in.

Reply:
@mtrmac @umohnani8 Should we proceed with this PR containing the note on the CRI-O logs reporting the full name of the image repository? Or hold until any code changes are made to improve the CRI-O logs?